src/NCalc/NCalcParser.cs

```diff
@@ -2587,7 +2587,7 @@ private NCalcParser.value_return value()
 				adaptor.AddChild(root_0, INTEGER35_tree);
 
 				DebugLocation(208, 14);
-				 try { retval.value = new ValueExpression(int.Parse((INTEGER35!=null?INTEGER35.Text:null))); } catch(System.OverflowException) { retval.value = new ValueExpression(long.Parse((INTEGER35!=null?INTEGER35.Text:null))); } 
+				 try { retval.value = new ValueExpression(int.Parse((INTEGER35!=null?INTEGER35.Text:null))); } catch(System.OverflowException) { retval.value = new ValueExpression((object)long.Parse((INTEGER35!=null?INTEGER35.Text:null))); } 

```

Comment:
is there a way to change this in the grammar file itself? this file is generated code and ideally shouldn't be changed manually

Reply:
Sorry I am not sure how. Which file should I look into?

Reply:
I think it's here: https://github.com/sklose/NCalc2/blob/e9f2d1b73f2e680a01c66eed1ade5958b43524b5/grammer/NCalc.g#L209

Reply:
Thanks @Tradioyes. Added the change